packages/discovery-provider/src/tasks/entity_manager/entities/muted_user.py

```diff
@@ -0,0 +1,73 @@
+from src.exceptions import IndexingValidationError
+from src.models.moderation.muted_user import MutedUser
+from src.tasks.entity_manager.utils import (
+    Action,
+    EntityType,
+    ManageEntityParameters,
+    copy_record,
+    validate_signer,
+)
+from src.utils.structured_logger import StructuredLogger
+
+logger = StructuredLogger(__name__)
+
+
+def validate_mute_user_tx(params: ManageEntityParameters):
+    muted_user_id = params.entity_id
+    validate_signer(params)
+    if (
+        params.action == Action.CREATE
+        and (params.user_id, muted_user_id)
+        in params.existing_records[EntityType.MUTED_USER.value]
+    ):
+        raise IndexingValidationError(
+            f"User {params.user_id} already muted user {muted_user_id}"
+        )
```

Comment:
I know you mentioned needing to add more validation but I feel like we should just go ahead and add them; should add errors for missing `user_id` and `muted_user_id`